### DISCUSSION TITLE
Report | Week 08

### DISCUSSION THREAD

**Discussion by tech-n1c** [SE-TINF22B6/time2shine#37] [Show and tell]
## Overview

Update for the SRS: now including non-functional requirements.

---

### SRS ([link](https://github.com/SE-TINF22B6/time2shine/blob/main/Documentation/srs.md))
We added the Utility tree and non-functional requirements.

Some non-functional requirements:
1. Firstly, performance is very important for the user in order to have a good time playing the game. Therefore the whole architecture needs to be constructed in a way to minimize waiting time for the user. If it was not a game it wouldn`t be crucial.
2. Availibilty is very important for that game. To be specified, it is necessary for the highscore to be available all the time. This means, that the infrastructure of the database needs to be there 24/7. 
3. There are lot of cheaters out there that want to get themselves an advantage. Therefore the game needs some good cheaters` detection. Therefore it is a key feature of our game

---

### Utility Tree

The latest version can be found [here](https://github.com/SE-TINF22B6/time2shine/blob/main/Documentation/Quality_Attribute_Scenarios.md).

![image](https://github.com/SE-TINF22B6/time2shine/assets/59262249/9cc54394-07d7-4c19-9b73-c69816437adb)

**Comment by AndresMasis** [comment]
Hi

I agree with you that performance is a very important non functional requirement for your project. Definitely, a deficient performance will make any gaming servince unsuccesful because the user will not want to use a slow game. So I also think that it is important for you to heavily in this aspect.

In addition, desire to get a recreation time can get at any moment, so I also think it was a good point to invest in availability. This indicates that you are really understanding the stakeholder. This aspect of getting correctly the user concerns is very important for software engineering. Good job.

Cheers from CookPit
